src/journal/views.py

```diff
@@ -889,10 +912,42 @@ def manage_issues(request, issue_id=None, event=None):
         'issue': issue,
         'form': form,
         'modal': modal,
+        'galley_form': galley_form,
     }
 
     return render(request, template, context)
 
+@editor_user_required
+def issue_galley(request, issue_id, delete=False):
+    issue = get_object_or_404(models.Issue, pk=issue_id)
+
+    if request.method == 'POST':
+        form = issue_forms.IssueGalleyForm(request.POST, request.FILES)
+        if 'delete' in request.POST:
+            issue_galley = get_object_or_404(models.IssueGalley, issue=issue)
+            issue_galley.delete()
+            messages.info(request, "Issue Galley Deleted")
+        elif form.is_valid():
+            uploaded_file = request.FILES["file"]
+            try:
+                issue_galley = models.IssueGalley.objects.get(issue=issue)
+                issue_galley.replace_file(uploaded_file)
+            except models.IssueGalley.DoesNotExist:
+                file_obj = files.save_file(
+                        request,
+                        uploaded_file,
+                        label=issue.issue_title,
+                        public=False
+                        *(models.IssueGalley.FILES_PATH, issue.pk)
+                )
+                models.IssueGalley.objects.create(
+                    file=file_obj,
+                    issue=issue,
+                )
+
+            messages.info(request, "Issue Galley Uploaded")
+
```

Comment:
If I attempt to upload a non-pdf it is duly rejected, however, as there is no handler for where `form.is_valid()` is `False` I don't get told why it was rejected, may be useful to add a handler here for that?

Reply:
Ah good point, the form does render the errors, but it is not immediately shown due to it being a modal. I'll add a message as well